tests/app/celery/test_process_ga4_measurement_task.py

```diff
@@ -0,0 +1,52 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from app.celery.exceptions import AutoRetryException
+from app.celery.process_ga4_measurement_tasks import post_to_ga4
+
+
+def test_it_post_to_ga4_with_valid_data(ga4_sample_payload):
+    # Patch the requests.post method to return a 200 status code.
+    with patch('app.celery.process_ga4_measurement_tasks.requests') as mock_requests:
+        mock_requests.post.return_value = MagicMock(status_code=204)
+        response = post_to_ga4(
+            ga4_sample_payload['notification_id'],
+            ga4_sample_payload['template_name'],
+            ga4_sample_payload['template_id'],
+            ga4_sample_payload['service_id'],
+            ga4_sample_payload['service_name'],
+            client_id=ga4_sample_payload['client_id'],
+            name=ga4_sample_payload['name'],
+            source=ga4_sample_payload['source'],
+            medium=ga4_sample_payload['medium'],
+        )
+
+    assert response
+
+
+def test_it_post_to_ga4_returns_4xx(ga4_sample_payload):
+    with patch('app.celery.process_ga4_measurement_tasks.requests.post') as mock_post:
+        mock_post.return_value.status_code = 400
+        response = post_to_ga4(
+            ga4_sample_payload['notification_id'],
+            ga4_sample_payload['template_name'],
+            ga4_sample_payload['template_id'],
+            ga4_sample_payload['service_id'],
+            ga4_sample_payload['service_name'],
+        )
+
+    assert not response
+
+
+def test_it_post_to_ga4_missing_config(ga4_sample_payload):
```

Comment:
It doesn't make sense to me that we would retry missing configuration because that condition will not self-resolve.